### ISSUE TITLE
feat: "Pay as you go" plan

### ISSUE THREAD

**Issue by GeylanKalafMohe** [shorebirdtech/shorebird#1187]
**Description**

‪I wish to be able to have an option like a "pay as you go"‬ plan. It would be easier for companies and individuals to pay the monthly subscription of shorebird. Especially when you have months when you do not plan to release a new path.

I am thinking of for example:

0.0003$ per patch as a complete price without the base subscription of 20$.
Maybe you have another better price plan which goes into this direction.

Thank you.

**Comment by eseidel** [comment]
Thank you for the suggestion.  We don't currently have the infrastructure to actually bill for overages, so existing account holders can essentially serve unlimited patches on current accounts (within reason, very large apps we have direct contact with and just have separate plans for).  We're working on adding per-patch billing infrastructure at which time we intend to start charging per-patch beyond the initial 50k for those who need such.  Once we have something like that working we could consider doing "pay as you go" or really allowing the existing free plan to have overages (right now a free plan will stop serving patches after about 5000 patches in a given month).  The major purpose of this pricing update was to lower our advertised prices and make Shorebird accessible to a wider set of budgets.  It's possible "pay as you go" could do similar, but we need to do some infrastructure work before we can consider such.  Thanks.

**Comment by eseidel** [comment]
Thanks for the suggestion.  We've made a few changes since this.  We've added more tiers at higher levels and have a generous free plan of 5000 patches.  It's now possible for a business to upgrade/downgrade their subscription to whatever level they need from console.shorebird.dev.  We do not currently support paying per-patch, we looked into it and doing after-the-fact per-patch billing was just too complicated to add at this time so we've stuck with monthly pre-paid tiers for now.

Thanks again for the feedback.
